app/controllers/topics_controller.rb

```diff
@@ -17,11 +20,31 @@ def create
   end
 
   def show
-    @topic = Topic.find(params[:id])
+  end
+
+  def destroy
+    Topic.find(params[:id]).destroy
+    redirect_to topics_path
+  end
+
+  def edit
+  end
+
+  def update
+    if @topic.update(topic_params)
+      redirect_to topic_path
+    else
+      @topic.name = Topic.find(params[:id]).name
+      render :edit
+    end
   end
 
   private
 
+  def topic_find
```

Comment:
Really minor naming point (no need to change)....
There are two types of methods: methods that return values (functions) and methods that have side effects. (In fact all methods return values, but for some methods you don't care about the value returned)

My rules for naming are:
- When a method returns a value (i.e. a function) name the method after the the thing being returned (e.g. "topic", "user", "tag_collection").
- When a method has a side effect name the method in the present tense imperative (e.g. "run away", "stop what you are doing" "sit down").

In this case the method has a side effect of setting an instance variable, so I'd name it `find_topic`